### PR TITLE
Docker deployment setup

### PR DESCRIPTION
**Pull request by conquext**
## Installation (On Local Machine)

Install dependencies (first time only):

    $ npm install

## Installation (Using Docker)

    $ docker build -t gads/customyz:1.0 .

How to start:

- On local machine
    $ node server

- On docker
    ```docker run -p 4567:8080 imageId```

- How to deploy docker image
    ```docker tag imageId conquext/customyz:gads01server```
    ```docker push conquext/customyz:gads```

Open [http://localhost:4567](http://localhost:4567)

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "customizr",
```
Comment (Oisavictor) [inline server/package.json:2]: shouldn't it be cuztomyz?

File: src/html/invoice.html [hunk 2]
```diff
@@ -0,0 +1,339 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>InvoiceS</title>
```
Comment (Oisavictor) [inline src/html/invoice.html]: I think it should be spelt like this "Invoice" and not "InvoiceS"?

File: src/html/invoice.html [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+            </div>
+            <div class="mb-2">
+              <label for="date">Payment Date:</label>
+              <input
+                type="date"
+                id="date"
+                class="form-control"
+                placeholder="mm/dd/yyyy"
+              />
+            </div>
```
Comment (Oisavictor) [inline src/html/invoice.html:158]: I suggested that the Payment type shouldn't be text and should be auto generated.
Also Payment date should be auto generated

File: server/firebase.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+const env = process.env;
+
+const config = {
+  apiKey: env.apiKey,
+  authDomain: env.authDomain,
+  databaseURL: env.databaseURL,
+  projectId: env.projectId,
+  storageBucket: env.storageBucket,
+  messagingSenderId: env.messagingSenderId,
+};
```
Comment (Oisavictor) [inline server/firebase.js:14]: @conquext why didn't you add the details i sent to you here? for security concerns?

File: server/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    $ docker build -t gads/customyz:1.0 .
+
+How to start:
+
+- On local machine
+  \$ node server
+
+- On docker
+  \$ docker run -p 4567:8080 imageId
```
Comment (Oisavictor) [inline server/README.md:23]: why did you use port 8080? instead of port 80 as your open port?

File: dist/invoice.html [hunk 6]
```diff
@@ -0,0 +1,339 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>InvoiceS</title>
```
Comment (Oisavictor) [inline dist/invoice.html]: I think you should use "Invoice" instead of "Invoices"

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+WORKDIR /app
+
+COPY server/package*.json ./
+
+RUN npm install
+
+COPY server/ .
+COPY dist/ ./public/
+
+ENV PORT=8080
```
Comment (Oisavictor) [inline Dockerfile:12]: port 8080? while not port 80?

**Review by Oisavictor — CHANGES_REQUESTED** [review]
kindly see my comments

File: server/firebase.js (hunk 4 above)
Comment (conquext) [reply server/firebase.js:14]: yeah, it is added in the environment variables

File: server/README.md (hunk 5 above)
Comment (conquext) [reply server/README.md:23]: Port could be any valid address. 8080 is a valid address.

File: dist/invoice.html (hunk 6 above)
Comment (conquext) [reply dist/invoice.html]: I merged the code from frontend so i can build a container image to showcase the current build. Comments related to html can be moved under their respective PRs.

**Review by Oisavictor — APPROVED** [review]

**Review by TOBECHI-GH — APPROVED** [review]
Good work

**Pull request merged**